judge/runguard.cc

```diff
@@ -851,91 +799,91 @@ void setrestrictions()
 	setlim(STACK);
 
 	if ( filesize!=RLIM_INFINITY ) {
-		verbose("setting filesize limit to %lu bytes",filesize);
+		logmsg(LOG_DEBUG, "setting filesize limit to {} bytes",filesize);
 		lim.rlim_cur = lim.rlim_max = filesize;
 		setlim(FSIZE);
 	}
 
 	if ( nproc!=RLIM_INFINITY ) {
-		verbose("setting process limit to %d",(int)nproc);
+		logmsg(LOG_DEBUG, "setting process limit to {}",(int)nproc);
 		lim.rlim_cur = lim.rlim_max = nproc;
 		setlim(NPROC);
 	}
 
 #undef setlim
 
 	if ( no_coredump ) {
-		verbose("disabling core dumps");
+		logmsg(LOG_DEBUG, "disabling core dumps");
 		lim.rlim_cur = lim.rlim_max = 0;
-		if ( setrlimit(RLIMIT_CORE,&lim)!=0 ) error(errno,"disabling core dumps");
+		if ( setrlimit(RLIMIT_CORE,&lim)!=0 ) die(errno,"disabling core dumps");
 	}
 
 	/* Put the child process in the cgroup */
 	const char *controllers[] = { "memory", nullptr };
 	if (cgroup_change_cgroup_path(cgroupname, getpid(), controllers) != 0) {
-		error(0, "Failed to move the process to the cgroup");
+		die(0, "Failed to move the process to the cgroup");
 	}
 
 	/* Run the command in a separate process group so that the command
 	   and all its children can be killed off with one signal. */
-	if ( setsid()==-1 ) error(errno,"setsid failed");
+	if ( setsid()==-1 ) die(errno,"setsid failed");
 
 	/* Set root-directory and change directory to there. */
 	if ( use_root ) {
 		/* Small security issue: when running setuid-root, people can find
 		   out which directories exist from error message. */
-		if ( chdir(rootdir)!=0 ) error(errno,"cannot chdir to `%s'",rootdir);
+		if ( chdir(rootdir)!=0 ) die(errno,"cannot chdir to `{}'",rootdir);
 
 		/* Get absolute pathname of rootdir, by reading it. */
 		char  cwd[PATH_MAX+1];
-		if ( getcwd(cwd,PATH_MAX)==nullptr ) error(errno,"cannot get directory");
+		if ( getcwd(cwd,PATH_MAX)==nullptr ) die(errno,"cannot get directory");
 		if ( cwd[strlen(cwd)-1]!='/' ) strcat(cwd,"/");
 
 		/* Canonicalize CHROOT_PREFIX. */
 		char *path;
 		if ( (path = (char *) malloc(PATH_MAX+1))==nullptr ) {
-			error(errno,"allocating memory");
+			die(errno,"allocating memory");
 		}
 		if ( realpath(CHROOT_PREFIX,path)==nullptr ) {
-			error(errno,"cannot canonicalize path '%s'",CHROOT_PREFIX);
+			die(errno,"cannot canonicalize path `{}'",CHROOT_PREFIX);
 		}
 
 		/* Check that we are within prescribed path. */
 		if ( strncmp(cwd,path,strlen(path))!=0 ) {
-			error(0,"invalid root: must be within `%s'",path);
+			die(0,"invalid root: must be within `{}'",path);
 		}
 		free(path);
 
-		if ( chroot(".")!=0 ) error(errno,"cannot change root to `%s'",cwd);
-		if ( chdir("/")!=0 ) error(errno,"cannot chdir to `/' in chroot");
+		if ( chroot(".")!=0 ) die(errno,"cannot change root to `{}'",cwd);
+		if ( chdir("/")!=0 ) die(errno,"cannot chdir to '/' in chroot");
```

Comment:
Nitpick: quoting was inconsistently changed compared to the line above (and possibly other places in the file).

Reply:
Done